packages/gatsby-plugin-utils/src/polyfill-remote-file/utils/get-request-headers-for-url.ts

```diff
@@ -0,0 +1,36 @@
+import url from "url"
+import path from "path"
+import importFrom from "import-from"
+import resolveFrom from "resolve-from"
+
+import type { Headers } from "got"
+import type { Store } from "gatsby"
+
+export function getStore(): Store {
+  const gatsbyPkgRoot = path.dirname(
+    resolveFrom(process.cwd(), `gatsby/package.json`)
+  )
```

Comment:
This is problematic, we should only do this when we don't have a choice. We should make the store a parameter for `addRemoteFilePolyfillInterface` so we can pass in this information to the resolver.

Gatsby can do the same and the gatsbyImageResolver and such can use it just like we do with actions.

Reply:
I made that change in https://github.com/gatsbyjs/gatsby/pull/35655/commits/dd9f607c6bee639157fec82f133a873522c0dfb6 but it feels like a more brittle way of doing it. Why can't we import the store? As of this change plugin-utils relies on the store now 🤔

Reply:
Oops, https://github.com/gatsbyjs/gatsby/pull/35655/commits/0f0588a6f6141e5f4b47a136066ad016c51c8652 was required as well

Reply:
I guess we will also need to make a PR to the source-dato repo now so it doesn't show warnings about the store missing because of this line https://github.com/gatsbyjs/gatsby/commit/dd9f607c6bee639157fec82f133a873522c0dfb6#diff-b3f42d8a431fa95ef88a60e63137203da5028dafde2138c181ce53e522e02723R12-R13

Reply:
https://github.com/gatsbyjs/gatsby/pull/35655/commits/0b8213507b0465355827b99e94e7c5c680a25d85 was needed as well